### PR TITLE
Enable torch deterministic config for mojo deterministic mode

### PR DESCRIPTION
**Pull request by byteyushikang**
## Summary

- Configure PyTorch deterministic behavior when `MOJO_DETERMINISTIC=1` is enabled.
- Add `configure_torch_deterministic()` to call `torch.manual_seed(0)` and `torch.use_deterministic_algorithms(True)`.
- Add focused tests for the deterministic algorithm flag and manual seed behavior.

## Validation

- `python -m pytest mojo_opset/tests/base/test_deterministic_config.py mojo_opset/tests/base/test_backend_dispatch.py`
- `git diff --check`
- `python -m py_compile mojo_opset/utils/misc.py mojo_opset/backends/__init__.py mojo_opset/tests/base/test_deterministic_config.py`

### CODE REVIEW

File: mojo_opset/utils/misc.py [hunk 1]
```diff
@@ -13,6 +13,13 @@ def get_bool_env(key, default=True):
         return default
 
 
+def configure_torch_deterministic(seed=0):
+    import torch
+
+    torch.manual_seed(seed)
+    torch.use_deterministic_algorithms(True)
```
[the remark this answers is not in the document]
Comment (byteyushikang) [reply mojo_opset/utils/misc.py:20]: 增加了cudnn的确定性配置，但是不设置CUBLAS_WORKSPACE_CONFIG

**Review by wwens7 — APPROVED** [review]
